### PR TITLE
 fix parse() throws error on valid ReflectOnly Blink extended attributes

### LINKED ISSUE #256: parse() throws error on valid ReflectOnly Blink extended attributes
According to the Blink IDL Extended Attributes spec, [ReflectOnly has two valid forms](https://chromium.googlesource.com/chromium/src/+/master/third_party/blink/renderer/bindings/IDLExtendedAttributes.md#ReflectOnly_a):

    [Reflect, ReflectOnly="on"] attribute DOMString toggle;
    // Parens may contain 2 to n values.
    [Reflect=q, ReflectOnly=("anonymous", "use-credentials")] attribute DOMString quarter;

The second format is causing error such as the following to be thrown:

    Got an error during or right after parsing `interface HTMLImageElement`: Expected
    identifiers but none found, line 33 (tokens: "\"anonymous\",\"use-credentials\"),")

There are 11 files in Chromium (listed below) that trigger this error. I've [already verified](https://bugs.chromium.org/p/chromium/issues/detail?id=920009) that the Blink IDL spec is correct.

blink/renderer/core/html/html_element.idl
blink/renderer/core/html/html_link_element.idl
blink/renderer/core/html/html_table_cell_element.idl
blink/renderer/core/html/html_anchor_element.idl
blink/renderer/core/html/forms/html_form_element.idl
blink/renderer/core/html/html_area_element.idl
blink/renderer/core/html/html_script_element.idl
blink/renderer/core/html/html_iframe_element.idl
blink/renderer/core/html/html_image_element.idl
blink/renderer/core/html/media/html_media_element.idl
blink/renderer/core/dom/element.idl

I would appreciate some haste in addressing this. This affects tooling I've built to increase my velocity at [documenting new web platform APIs](https://developer.mozilla.org/en-US/profiles/jpmedley). If I can assist in some further way, will be happy to do so.

Thanks

### PR DESCRIPTION
**Pull request by TG1999**
Signed-off-by: tushar goel <tushar.goel.dav@gmail.com>

This patch closes #256 and includes:
- [ ] A relevant test
- [ ] A relevant documentation update

### CODE REVIEW

**Comment by TG1999** [comment]
@saschanaz please have a look on it :grinning:

File: lib/productions/helpers.js [hunk 1]
```diff
@@ -289,3 +289,25 @@ export function autoParenter(data, parent) {
     }
   });
 }
+
+/**
+ * Returns a proxy that auto-assign `parent` field.
+ * @template T
+ * @param {T} tokeniser
+ * @param {*} [tokenName] either can be a string or identifier
+ * @return {T}
```
[the remark this answers is not in the document]
Comment (TG1999) [reply lib/productions/helpers.js]: Okay got it

**Comment by TG1999** [comment]
How can I merge the changes that you done in gh-pages @saschanaz

**Comment by saschanaz** [comment]
> How can I merge the changes that you done in gh-pages @saschanaz

That worked! Though the new (expected) error:

>##[error]  313:2  error  Newline required at end of file but not found  eol-last

File: lib/productions/helpers.js [hunk 2]
```diff
@@ -43,9 +42,7 @@ export function list(tokeniser, { parser, allowDangler, listName = "list" }) {
   return items;
 }
 
-/**
- * @param {import("../tokeniser").Tokeniser} tokeniser
- */
```
Comment (saschanaz) [inline lib/productions/helpers.js]: Oops, this one is still needed!

File: lib/productions/extended-attributes.js [hunk 3]
```diff
@@ -1,8 +1,41 @@
...
 import { ArrayBase } from "./array-base.js";
-import { list, identifiersOrStrings, argument_list, autoParenter } from "./helpers.js";
+import { Token } from "./token.js";
+import { list, argument_list, autoParenter } from "./helpers.js";
 import { validationError } from "../error.js";
 
+/**
+ * @param {import("../tokeniser").Tokeniser} tokeniser
+ */
+function identifiers(tokeniser) {
```
Comment (saschanaz) [inline lib/productions/extended-attributes.js]: Oops, forgot that we are replacing `identifiers` with `identifiersOrStrings`. That means it's safe to remove this!

**Comment by saschanaz** [comment]
Closing in favor of #445

**Pull request closed without merging**